lib/docker/template/cache.rb

```diff
@@ -45,9 +43,10 @@ def aliased_context(builder)
       # ----------------------------------------------------------------------
 
       def cleanup(repo)
+        return unless repo.clean_cache?
```

Comment:
The change you made won't fix that as it's the same thing just as a short-circuit that provides no enhancement.